src/kixi/datastore/web_server.clj

```diff
@@ -159,6 +165,20 @@
                 (ms/retrieve metadatastore id)
                 (return-unauthorised ctx))))}}})
 
+(defn file-download
+  [metrics metadatastore filestore]
+  {:id :file-download
+   :methods
+   {:get {:produces [{:media-type #{"application/octet-stream"}}]
+          :response
+          (fn [ctx]
+            (let [id (get-in ctx [:parameters :path :id])]
+              (if (ms/authorised metadatastore ::ms/file-read id (ctx->user-groups ctx))
+                (let [md (ms/retrieve metadatastore id)
+                      filename (str (::ms/name md) "." (::ms/file-tye md))]
```

Comment:
I'm not super happy with it, but it's a side effect of the freeform file-type we adopted the other day. If we can get a white-list/black-list in place and have it spec'd then i think it should be ok